pkg/apis/core/validation/validation.go

```diff
@@ -2931,6 +2933,18 @@ func validateExecAction(exec *core.ExecAction, fldPath *field.Path) field.ErrorL
 	return allErrors
 }
 
+func validateSleepAction(sleep *core.SleepAction, terminationGracePeriodSeconds *int64, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+	gracePeriodSeconds := int64(30)
```

Comment:
This seems like a magic number, but I can't find where we use the default terminationGracePeriodSeconds😢

Reply:
30s is the default terminationGracePeriodSeconds

Reply:
You don't need to handle nil.  tGPS is always set - if the user did not set a value, we default it.  Just read the value, no hard-coded default  here (which is WHY we default the value :)

You should be able to pass around the grace period as int64 rather than pointer

Reply:
Understood now, but I think we still need to handle nil here, as many spec with tGPS set to nil is used in tests. I removed the hard coded default though.